server/util/lru/lru.go

```diff
@@ -30,16 +32,17 @@ type Config struct {
 type LRU struct {
 	sizeFn      SizeFn
 	evictList   *list.List
-	items       map[uint64]*list.Element
+	items       map[uint64][]*list.Element
```

Comment:
nit: might be able to get away with making the key a `struct{uint64, uint64}` -- then you don't need any list iteration logic

Reply:
(only issue I can think of is that it increases the memory used by the keys, but you'd also be saving on the extra memory required for the slice values)